AsyncDisplayKit/ASCollectionView.h

```diff
@@ -26,29 +26,59 @@ NS_ASSUME_NONNULL_BEGIN
 /**
  * Asynchronous UICollectionView with Intelligent Preloading capabilities.
  *
- * ASCollectionNode is recommended over ASCollectionView.  This class exists for adoption convenience.
- *
- * ASCollectionView is a true subclass of UICollectionView, meaning it is pointer-compatible
+ * @discussion ASCollectionView is a true subclass of UICollectionView, meaning it is pointer-compatible
  * with code that currently uses UICollectionView.
  *
  * The main difference is that asyncDataSource expects -nodeForItemAtIndexPath, an ASCellNode, and
  * the sizeForItemAtIndexPath: method is eliminated (as are the performance problems caused by it).
  * This is made possible because ASCellNodes can calculate their own size, and preload ahead of time.
+ *
+ * @note ASCollectionNode is strongly recommended over ASCollectionView.  This class exists for adoption convenience.
  */
 @interface ASCollectionView : UICollectionView
 
 /**
- * Initializer.
+ * Initializes an ASCollectionView
+ *
+ * @discussion Initializes and returns a newly allocated collection view object with the specified layout.
  *
  * @param layout The layout object to use for organizing items. The collection view stores a strong reference to the specified object. Must not be nil.
  */
 - (instancetype)initWithCollectionViewLayout:(UICollectionViewLayout *)layout;
+
+/**
+ * Initializes an ASCollectionView
+ *
+ * @discussion Initializes and returns a newly allocated collection view object with the specified frame and layout.
+ *
+ * @param frame The frame rectangle for the collection view, measured in points. The origin of the frame is relative to the superview in which you plan to add it. This frame is passed to the superclass during initialization.
+ * @param layout The layout object to use for organizing items. The collection view stores a strong reference to the specified object. Must not be nil.
+ */
 - (instancetype)initWithFrame:(CGRect)frame collectionViewLayout:(UICollectionViewLayout *)layout;
 
-// The corresponding ASCollectionNode, which exists even if directly allocating & handling the view class.
+/**
+ * Returns the corresponding ASCollectionNode
+ *
+ * @return collectionNode The corresponding ASCollectionNode which exists even if directly allocating & handling the view class.
+ */
 @property (nonatomic, weak, readonly) ASCollectionNode *collectionNode;
 
+/**
+ * The object that acts as the asynchronous delegate of the collection view
+ *
+ * @discussion The delegate must adopt the ASCollectionDelegate protocol. The collection view maintains a weak reference to the delegate object.
+ *
+ * The delegate object is responsible for providing size constraints for nodes and indicating whether batch fetching should begin.
+ */
 @property (nonatomic, weak) id<ASCollectionDelegate>   asyncDelegate;
+
+/**
+ * The object that acts as the asynchronous data source of the collection view
+ *
```

Comment:
@garrettmoon could you start a -core thread on header doc style?  My view is that we should densify these more, especially when there is not a lot of content / the method is simple, even if it creates a minor inconsistency.  E.g. for this comment we could use between 2-3 fewer lines.